File: java/Samples/Biometrics/CS/AbisSampleCS/Common/TabPageContentBase.cs
using System.ComponentModel;
using System.Windows.Forms;

namespace Neurotec.Samples
{
	public partial class TabPageContentBase : UserControl, INotifyPropertyChanged
	{
		#region Public constructor

		public TabPageContentBase()
		{
			InitializeComponent();
		}

		#endregion

		#region Private fields

		private string _tabName = "Tab";

		#endregion

		#region Public properties

		public ITabController TabController { get; set; }

		public string TabName
		{
			get { return _tabName; }
			set
			{
				if (_tabName != value)
				{
					_tabName = value;
					if (PropertyChanged != null)
					{
						PropertyChanged(this, new PropertyChangedEventArgs("TabName"));
					}
				}
			}
		}

		#endregion

		#region Public virtual methods

		public virtual void OnTabAdded()
		{
		}

		public virtual void OnTabEnter()
		{
		}

		public virtual void OnTabLeave()
		{
		}

		public virtual void OnTabClose()
		{
		}

		public virtual void SetParams(params object[] parameters)
		{
		}

		#endregion

		#region INotifyPropertyChanged Members

		public event PropertyChangedEventHandler PropertyChanged;

		#endregion
	}
}

File: java/Samples/Biometrics/VB/AbisSampleVB/Common/TabPageContentBase.vb
Imports Microsoft.VisualBasic
Imports System.Windows.Forms
Imports System.ComponentModel
Imports Neurotec.Biometrics.Client
Imports System

Partial Public Class TabPageContentBase
	Inherits UserControl
	Implements INotifyPropertyChanged
	#Region "Public properties"

	Public Sub New()
		InitializeComponent()
	End Sub

	#End Region

	#Region "Private fields"

	Private _tabName As String = "Tab"

	#End Region

	#Region "Public properties"

	Private privateTabController As ITabController
	Public Property TabController() As ITabController
		Get
			Return privateTabController
		End Get
		Set(ByVal value As ITabController)
			privateTabController = value
		End Set
	End Property

	Public Property TabName() As String
		Get
			Return _tabName
		End Get
		Set(ByVal value As String)
			If _tabName <> value Then
				_tabName = value
				RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs("TabName"))
			End If
		End Set
	End Property

	#End Region

	#Region "Public virtual methods"

	Public Overridable Sub OnTabAdded()
	End Sub

	Public Overridable Sub OnTabEnter()
	End Sub

	Public Overridable Sub OnTabLeave()
	End Sub

	Public Overridable Sub OnTabClose()
	End Sub

	Public Overridable Sub SetParams(ParamArray ByVal parameters() As Object)
	End Sub

	#End Region

	#Region "INotifyPropertyChanged Members"

	Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

	#End Region
End Class
